crypto/evp/mac_lib.c

```diff
@@ -48,18 +48,19 @@ void EVP_MAC_CTX_free(EVP_MAC_CTX *ctx)
     OPENSSL_free(ctx);
 }
 
-int EVP_MAC_CTX_copy(EVP_MAC_CTX *dst, EVP_MAC_CTX *src)
+EVP_MAC_CTX *EVP_MAC_CTX_dup(const EVP_MAC_CTX *src)
 {
-    EVP_MAC_IMPL *macdata;
+    EVP_MAC_CTX *dst = EVP_MAC_CTX_new(src->meth);
 
-    if (src->data != NULL && !dst->meth->copy(dst->data, src->data))
-        return 0;
+    if (dst == NULL)
+        return NULL;
 
-    macdata = dst->data;
-    *dst = *src;
-    dst->data = macdata;
+    if (src->data != NULL && src->meth->dup(dst->data, src->data) == 0) {
```

Comment:
Because EVP_MAC is a new type, I think you can freely change the dup function to be a real dup